00-programming-fundamentals/setting-up-vs-code-debugger.md

```diff
@@ -0,0 +1,112 @@
+# Setting up the VSCode Debugger
+
+## Learning Goals
+- Be able to explain the purpose of an Integrated Development Environment
+- Understand how to set up an individual .rb file for debugging in VS
+
+## Text Editors vs. IDEs
+Whenever you go to write code, you have options about the tools that you use to employ. These can range from the most basic text editors like [Vim](https://www.vim.org/) and [Emacs](https://www.gnu.org/software/emacs/) to development tools fully integrated with specific languages like [PyCharm](https://www.jetbrains.com/pycharm/) or [WebStorm](https://www.jetbrains.com/webstorm/), or something in between like [Atom](https://atom.io/), or our very own VS Code.
+
+For every two developers, there are three opinions about the best environment for making code. So how do we pick? It all depends on how much help (some would say interference) a programmer wants moment to moment.
+
+On the most basic level, a Text Editor will do just that: let you edit the words and symbols on a page unfettered by suggestion text or drop downs. An Integrated Development Environment (IDE for short) on the other hand has opinions about how you should be doing your work: they may try and give you the name of the function you were about to type, help manage a source control tool, or (as we will do) help you debug your code.
+
+__Note: If anyone ever tells you that "real developers only use [their personal favorite tool]", laugh in their face and walk away from that conversation.__
+
+## Setting Up
+The relationship between VS Code and Ruby is still in its infancy, so many of the tools that we regularly use in other languages (such as C++, Javascript or Ook) aren't immediately accessible. Our process will be:
+
+1. Set up the file we want to debug
```

Comment:
This step (and the `my_efficient_and_clever_program.rb`) doesn't seem to be relevant to this setup exercise, is it possible to move this step? Basically, the code snippet below doesn't get used until the next activity, and it may be good to keep those bundled together.

Perhaps there are steps for setting up, and then steps outlined in the next activity includes "set up the file we want to debug."